src/interval.jl

```diff
@@ -171,6 +171,22 @@ function Base.last(interval::Interval{T,L,R}) where {T,L,R}
     return R !== Unbounded ? interval.last : nothing
 end
 
+function Base.minimum(interval::Interval{T,Closed,R}; precision=nothing) where {T,R}
+    return first(interval)
+end
+
+function Base.minimum(interval::Interval{T,Open,R}; precision=eps(T)) where {T,R}
+    return first(interval) + precision
+end
+
```

Comment:
You can make things nicer for integers by doing:
```suggestion

function Base.minimum(interval::Interval{T,L,R}; precision=one(T)) where {T <: Integer, L, R}
    return minimum(interval; precision=precision)
end

```

Reply:
We should also probably special case `AbstractFloat` as with very large floating-point values adding `eps` is inaccurate:
```julia
julia> x = nextfloat(typemin(Float64))
-1.7976931348623157e308

julia> x + eps(Float64)
-1.7976931348623157e308

julia> x + eps(Float64) == x
true

julia> nextfloat(x)
-1.7976931348623155e308

julia> nextfloat(x) == x
false
```